Eastern-Research-Group/mywaterway
# Feature/558 monitoring report line graph

## Related Issues:
* [HMW-558](https://jira.epa.gov/browse/HMW-558)

## Main Changes:
* Added the option to chart the measurements on the Monitoring Report page as line graphs rather than scatter plots.
  * To differentiate between different depths, multiple lines are used on the line graph. The colors of the lines are interpolated along a simple gradient for a base hue. This interpolation is determined per chart, since I didn't want to make assumptions about possible depths. I decided to use this gradient for the scatterplot, too, but we may want to see what EPA thinks.
  * This depends on the assumption that multiple measurements in a single day are for different depths. I think this has been established, but it should be confirmed. If this isn't the case, then multiple measurements for a single depth in the same day will need to be averaged for the line graph.
  * The colors don't raise contrast issues with accessibility tools, but some may be subjectively too light. It may be necessary to further limit the range of the gradient or add a slight background to the chart.
* Fixed some lingering Visx chart and style issues.

## Steps To Test:
1. Go to http://localhost:3000/monitoring-report/STORET/CBP_WQX/CBP_WQX-ANA24/.
2. In the table, select the characteristics **Chlorophyll a** and **Temperature, water**.
3. Scroll down, and confirm that lighter colored dots on the scatter plots correspond to shallower depths.
4. Switch to the line graph in both charts.
5. Confirm the data is plotted as expected. There will be breaks in some of the lines.

## TODO:
- [ ] Confirm with EPA/WQP that multiple measurements in a single day correspond to different depths.
